File: src/Logitar.Portal.Contracts/Configurations/Configuration.cs
using Logitar.Portal.Contracts.Settings;

namespace Logitar.Portal.Contracts.Configurations;

public record Configuration : Aggregate
{
  public Locale DefaultLocale { get; set; } = new();
  public string Secret { get; set; } = string.Empty;

  public UniqueNameSettings UniqueNameSettings { get; set; } = new();
  public PasswordSettings PasswordSettings { get; set; } = new();

  public LoggingSettings LoggingSettings { get; set; } = new();
}

File: src/Logitar.Portal.Ui/src/types/configuration.ts
import type { Aggregate } from "@/types/aggregate";
import type { CustomAttribute } from "@/types/customAttributes";
import type { PasswordSettings, UniqueNameSettings } from "@/types/settings";
import type { Session } from "@/types/sessions";

export type Configuration = Aggregate & {
  defaultLocale: string;
  secret: string;
  uniqueNameSettings: UniqueNameSettings;
  passwordSettings: PasswordSettings;
  loggingSettings: LoggingSettings;
};

export type InitializeConfigurationPayload = {
  locale: string;
  user: UserPayload;
  session?: SessionPayload;
};

export type InitializeConfigurationResult = {
  configuration: Configuration;
  session: Session;
};

export type IsConfigurationInitializedResult = {
  isInitialized: boolean;
};

export type LoggingSettings = {
  extent: string;
  onlyErrors: boolean;
};

export type SessionPayload = {
  isPersistent: boolean;
  customAttributes: CustomAttribute[];
};

export type UpdateConfigurationPayload = {
  defaultLocale?: string;
  secret?: string;
  uniqueNameSettings?: UniqueNameSettings;
  passwordSettings?: PasswordSettings;
  loggingSettings?: LoggingSettings;
};

export type UserPayload = {
  uniqueName: string;
  password: string;
  emailAddress: string;
  firstName: string;
  lastName: string;
};
